qa/rpc-tests/mempool_tx_expiry.py

```diff
@@ -0,0 +1,205 @@
+
+        # Verify transparent transaction is version 3 intended for Overwinter branch
+        rawtx = self.nodes[0].getrawtransaction(persist_transparent, 1)
+        assert_equal(rawtx["version"], 3)
+        assert_equal(rawtx["overwintered"], True)
+        assert_equal(rawtx["expiryheight"], 211)
+        print "Blockheight at persist_transparent & persist_shielded creation:", self.nodes[0].getinfo()['blocks']
+        print "Expiryheight of persist_transparent:", rawtx['expiryheight']
+
+        # Verify shielded transaction is version 3 intended for Overwinter branch
+        rawtx = self.nodes[0].getrawtransaction(persist_shielded, 1)
+        print "persist_shielded expiryheight", rawtx['expiryheight']
+        print "Expiryheight of persist_shielded", rawtx['expiryheight']
+        assert_equal(rawtx["version"], 3)
+        assert_equal(rawtx["overwintered"], True)
+        assert_equal(rawtx["expiryheight"], 211)
+
+        ### Test before, at, and after expiry block ###
+
+        # Before expiry block height, txs get mined on node 0
+        assert(persist_transparent in self.nodes[0].getrawmempool())
+        assert(persist_shielded in self.nodes[0].getrawmempool())
+        # Txs are not on node 2
+        assert_equal(set(self.nodes[2].getrawmempool()), set())
+        print "persist_shielded: ", persist_shielded
+        print "mempool node 0:", self.nodes[0].getrawmempool()
+        print "mempool node 2:", self.nodes[2].getrawmempool()
+        bal = self.nodes[0].z_gettotalbalance()
+        print "Printing balance before persist_shielded & persist_transparent are initially mined from mempool", bal
+        # Txs are mined on node 0; will later be rolled back
+        self.nodes[0].generate(1)
+        print "Node 0 generated 1 block"
+        print "Node 0 height:", self.nodes[0].getinfo()['blocks']
+        print "Node 2 height:", self.nodes[2].getinfo()['blocks']
+        # Check private balance after persist_shielded is mined on node 0
+        bal = self.nodes[0].z_gettotalbalance()
+        print "Printing balance after persist_shielded & persist_transparent are mined:", bal
+        # Check that txs have been removed from node 0
+        assert(persist_transparent not in self.nodes[0].getrawmempool())
+        assert(persist_shielded not in self.nodes[0].getrawmempool())
+
+        ## BEFORE expiry block height: Reorg and test case where txs persist in mempool
+        print "Mine 2 competing blocks on Node 2..."
+        blocks = self.nodes[2].generate(2)
+        for block in blocks:
+            blk = self.nodes[2].getblock(block)
+            print "Height: {0}, Mined block txs: {1}".format(blk["height"], blk["tx"])
+        print "Connect nodes to force a reorg"
+        connect_nodes_bi(self.nodes,0,2)
+        self.is_network_split = False
+
+        print "Syncing blocks"
+        sync_blocks(self.nodes)
+
+        print "Ensure that txs are back in mempool of both nodes now"
```

Comment:
The test doesn't actually check this; it only checks they are in node 0's mempool.